### PR TITLE
Add search and metrics healthchecks for content-data-api

### PR DESCRIPTION
**Pull request by brucebolt**
In https://github.com/alphagov/content-data-api/pull/1595, we added `/healthcheck/search` and `/healthcheck/metrics` endpoints which will allow us to decouple application metrics from instance metrics.  As these metrics do not indicate a healthy instance, they should not be used in load balancing (which they are at present).

This adds these as alerts in Icinga, so we can soon remove the search and metrics from the main application healthcheck.

Trello card: https://trello.com/c/X2AKAUKn and https://trello.com/c/aDoOuYf5

### CODE REVIEW

File: modules/monitoring/manifests/checks.pp [hunk 1]
```diff
@@ -58,6 +59,36 @@
 
   include icinga::plugin::check_http_timeout_noncrit
 
+  $content_data_api_hostname    = "content-data-api.${app_domain}"
+  $content_data_api_metrics_url = '/healthcheck/metrics'
+  $content_data_api_search_url = '/healthcheck/search'
```
Comment (ChrisBAshton) [inline modules/monitoring/manifests/checks.pp:64]: Is it worth renaming these from `_url` to `_path`?

File: modules/monitoring/manifests/checks.pp [hunk 2]
```diff
@@ -58,6 +59,36 @@
...
+
+  if $::aws_migration {
+    include icinga::client::check_json_healthcheck
+
+    icinga::check { "content_data_api_search_${::hostname}":
+      check_command              => "check_app_health!check_json_healthcheck!443 ${content_data_api_search_url} ${content_data_api_hostname} https",
+      service_description        => 'search healthcheck for content-data-api',
+      use                        => 'govuk_urgent_priority',
+      host_name                  => $::fqdn,
+      notes_url                  => monitoring_docs_url(content-data-api-app-healthcheck-not-ok),
```
Comment (ChrisBAshton) [inline modules/monitoring/manifests/checks.pp:74]: I guess this makes sense for now: https://docs.publishing.service.gov.uk/manual/alerts/content-data-api-app-healthcheck-not-ok.html

Should we have a follow-up PR here and in the Dev Docs to split out into two notes_url pages?

**Review by ChrisBAshton — APPROVED** [review]
LGTM 👍 no blocking suggestions.

File: modules/monitoring/manifests/checks.pp (hunk 1 above)
Comment (brucebolt) [reply modules/monitoring/manifests/checks.pp]: Yes, that seems like a good idea 👍

File: modules/monitoring/manifests/checks.pp (hunk 2 above)
Comment (brucebolt) [reply modules/monitoring/manifests/checks.pp:74]: Indeed.  I'd planned to update the documentation as part of https://trello.com/c/Wn4w66dE, prior to removing the alerts from the instance healthcheck.

**Pull request merged**